system_setup/server/controllers/locale.py

```diff
@@ -30,3 +32,65 @@ def start(self):
                           .format(self.model.selected_language))
 
         super().start()
+
+    async def _run_locale_support_cmds(self, lang: str):
+        """ Final commands to run when a valid language is POST'ed. """
+        cmds = [["locale-gen"],
+                ["update-locale", "LANG={}".format(lang)]]
+        env = os.environ.copy()
+
+        cp = await arun_command(["check-language-support", "-l", lang[0:5]],
+                                env=env)
+        if cp.returncode == 0:
+            packages = cp.stdout.strip('\n').split(' ')
+            cmds.append(["apt", "install", "-y"] + packages)
+
+        if self.app.opts.dry_run:
+            for cmd in cmds:
+                log.debug('Would run: ' + ' '.join(cmd))
+        else:
+            for cmd in cmds:
+                cp = await arun_command(cmd, env=env)
+                if cp.returncode:
+                    log.debug('Command \"%s\" failed with return code %d',
```

Comment:
In the error case, that should rather be .error(), no?

Also, can we return the error to the caller which will handle it?

Reply:
Changed to an exception. It would really be an exceptional situation running that piece of code as non-root and no --dry-run simultaneosly outside of a dev environment.